airflow/providers/facebook/ads/operators/ads.py

```diff
@@ -0,0 +1,108 @@
+from airflow.providers.facebook.ads.hooks.ads import FacebookAdsReportingHook
+from airflow.providers.google.cloud.hooks.gcs import GCSHook
+from airflow.utils.decorators import apply_defaults
+
+
+class FacebookAdsReportToGcsOperator(BaseOperator):
+    """
+    Fetches the results from the Facebook Ads API as desired in the params
+    Converts and saves the data as a temporary JSON file
+    Uploads the JSON to Google Cloud Storage
+
+    .. seealso::
+        For more information on the Facebook Ads API, take a look at the API docs:
+        https://developers.facebook.com/docs/marketing-apis/
+
+    .. seealso::
+        For more information on the Facebook Ads Python SDK, take a look at the docs:
+        https://github.com/facebook/facebook-python-business-sdk
+
+    :param bucket: The GCS bucket to upload to
+    :type bucket: str
+    :param obj: GCS path to save the object. Must be the full file path (ex. `path/to/file.txt`)
+    :type obj: str
+    :param gcp_conn_id: Airflow Google Cloud Platform connection ID
+    :type gcp_conn_id: str
+    :param facebook_ads_conn_id: Airflow Facebook Ads connection ID
+    :type facebook_ads_conn_id: str
+    :param api_version: The version of Facebook API. Default to v6.0
+    :type api_version: str
+    :param fields: List of fields that is obtained from Facebook. Found in AdsInsights.Field class.
+        https://developers.facebook.com/docs/marketing-api/insights/parameters/v6.0
+    :type fields: List[str]
+    :param params: Parameters that determine the query for Facebook
+        https://developers.facebook.com/docs/marketing-api/insights/parameters/v6.0
+    :type params: Dict[str, Any]
+    :param gzip: Option to compress local file or file data for upload
+    :type gzip: bool
+    """
+
+    template_fields = ("facebook_ads_conn_id", "bucket_name", "object_name")
+
+    @apply_defaults
+    def __init__(
+        self,
+        bucket_name: str,
+        object_name: str,
+        gcp_conn_id: str = "google_cloud_default",
+        facebook_ads_conn_id: str = "facebook_default",
+        api_version: str = "v6.0",
+        fields: Optional[List[str]] = None,
+        params: Optional[Dict[str, Any]] = None,
+        gzip: bool = False,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.bucket_name = bucket_name
+        self.object_name = object_name
+        self.gcp_conn_id = gcp_conn_id
+        self.facebook_ads_conn_id = facebook_ads_conn_id
+        self.api_version = api_version
+        self.fields = fields
+        self.params = params  # type:ignore
+        self.gzip = gzip
+
+    def execute(self, context: Dict):
+        service = FacebookAdsReportingHook(facebook_ads_conn_id=self.facebook_ads_conn_id,
+                                           api_version=self.api_version)
+        rows = service.bulk_facebook_report(params=self.params, fields=self.fields)
+
+        converted_rows = [dict(row) for row in rows]
```

Comment:
Do you think that it is worth adding the option to save in CSV format? This format is very often needed as an input format for various older tools.

Reply:
yep cool, we can pass it as a arg! let the user decide.

Reply:
Can you restore the ability to save as JSON? I think CSV and JSON are very useful and should be the option of choice.

Here is an example of the last operator I did:
https://github.com/apache/airflow/blob/master/airflow/providers/google/cloud/example_dags/example_presto_to_gcs.py#L135